core/server/common/src/main/java/alluxio/master/journal/ufs/UfsJournal.java

```diff
@@ -189,7 +189,11 @@ public synchronized JournalContext createJournalContext() throws UnavailableExce
       // We throw UnavailableException here so that clients will retry with the next primary master.
       throw new UnavailableException("Failed to write to journal: journal is in state " + mState);
     }
-    return new MasterJournalContext(mAsyncWriter);
+    AsyncJournalWriter writer = mAsyncWriter;
+    if (writer == null) {
```

Comment:
This happens if any thread tries to write to a journal after it has been shutdown(e.g. due to loss of primacy). It should throw an `UnavailableException` per interface instead of a NPE.